code/modules/shuttles/shuttle_created.dm

```diff
@@ -0,0 +1,9 @@
+/datum/shuttle/autodock/overmap/created
+	defer_initialisation = TRUE
+
+/datum/shuttle/autodock/overmap/created/New(map_hash, obj/effect/shuttle_landmark/initial_location, list/initial_areas, tag)
+	name = tag
+	. = ..()
+	// Because these are created at runtime, we manually add the shuttle areas to the subsystem.
+	shuttle_area = initial_areas
+	SSshuttle.shuttle_areas += shuttle_area
```

Comment:
I don't like how you are not calling `SSshuttle.initialize_shuttle` on this thing. It means the shuttle added event won't fire, the SS now acts inconsistently, you have to duplicate misc code here, etc.